xc/device.ml

```diff
@@ -3001,9 +3007,14 @@ module Backend = struct
             ; Config.XenPlatform.device ~xs ~domid ~info
             ] in
 
-        let disks_cdrom, disks_other =
-          info.Dm_Common.disks
-          |> List.partition (function (_, _, Dm_Common.Cdrom) -> true | (_, _, _) -> false)
+        let disks_cdrom, disks_floppy, disks_other =
+          let acc (cdroms, floppies, other) ((_, _, media) as next) =
+            match media with
+            | Dm_Common.Cdrom  -> (next :: cdroms, floppies, other)
+            | Dm_Common.Floppy -> (cdroms, next :: floppies, other)
+            | Dm_Common.Disk   -> (cdroms, floppies, next :: other)
+          in
+          List.fold_left acc ([], [], []) info.Dm_Common.disks
```

Comment:
The fold reverses the order of the elements in the resulting lists. Not sure if it's relevant.

I recommend to change the name from `acc` to something like `partition`.

Reply:
hopefully not, all this function does is create a list of args for qemu

partition is indeed a sensible name